pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -359,6 +359,14 @@ func (gc *GarbageCollector) isDangling(reference metav1.OwnerReference, item *no
 		gc.absentOwnerCache.Add(reference.UID)
 		return true, nil, nil
 	}
+	if item.identity.Namespace == "" && owner.Namespace != "" {
+		klog.Infof("object %s is cluster scoped but its owner is in %s namespace", item.identity.UID, owner.Namespace)
+		return true, owner, nil
+	}
+	if item.identity.Namespace != "" && owner.Namespace != "" && owner.Namespace != item.identity.Namespace {
+		klog.Infof("object %s is in %s namespace but its owner is in %s namespace", item.identity.UID, item.identity.Namespace, owner.Namespace)
+		return true, owner, nil
+	}
```

Comment:
I don't think this check fixes the problem, since the mismatched namespace would cause the earlier [IsNotFound case](https://github.com/kubernetes/kubernetes/blob/534051acec00ab0dcaea502cc3bf410ba32c7b27/pkg/controller/garbagecollector/garbagecollector.go#L349) to get hit, adding the uid to the absentOwnerCache, and returning before this check would ever be reached

Reply:
In the original description, @caesarxuchao said case d (mismatching namespaces) is not covered.
I wonder if the description is accurate.

Reply:
It is accurate. mismatching namespaces is not covered.

Reply:
But the bug is specifically for correctly handling when an object has an owner reference which isn't meant to be covered.